File: LeetCode/0034-find-first-and-last-position-of-element-in-sorted-array/solution.java
class Solution {
    public int[] searchRange(int[] nums, int target) {
        int[] result = new int[2];
        result[0]=findFirst(nums,target);
        result[1]=findLast(nums,target);
        return result;
    }

    public static int findFirst(int[] nums, int target){
        int start = 0;
        int end = nums.length - 1;
        int mid = start + (end - start)/2;
        int ans = -1;

        while(start <= end){
            if(nums[mid] == target ){
                ans = mid;
                end = mid-1;
            }else if(nums[mid] > target){
                end = mid - 1;
            }else if(nums[mid] < target){
                start = mid + 1;
            }
            mid = start + (end - start)/2;
        }
        return ans;
    }

    public static int findLast(int[] nums, int target){
        int start = 0;
        int end = nums.length - 1;
        int mid = start + (end - start)/2;
        int ans = -1;

        while(start <= end){
            mid = start + (end - start)/2;
            if(nums[mid] == target){
                ans = mid;
                start = mid + 1;
            }else if(nums[mid] > target){
                end = mid - 1;
            }else if(nums[mid] < target){
                start = mid + 1;
            }
            mid = start + (end - start)/2;
        }
        return ans;
    }
}

File: LeetCode/0034-find-first-and-last-position-of-element-in-sorted-array/solution.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var searchRange = function(nums, target) {
    return [findFirst(nums,target),findLast(nums,target)]
};
function findFirst(nums,target){
        let start = 0;
        let end = nums.length - 1;
        let mid = Math.floor(start + (end - start)/2);
        let ans = -1;

        while(start <= end){
            if(nums[mid] == target ){
                ans = mid;
                end = mid-1;
            }else if(nums[mid] > target){
                end = mid - 1;
            }else if(nums[mid] < target){
                start = mid + 1;
            }
            mid = Math.floor(start + (end - start)/2);
        }
        return ans;
    };

    function findLast(nums, target){
        let start = 0;
        let end = nums.length - 1;
        let mid ;
        let ans = -1;

        while(start <= end){
            mid = Math.floor(start + (end - start)/2);
            if(nums[mid] == target){
                ans = mid;
                start = mid + 1;
            }else if(nums[mid] > target){
                end = mid - 1;
            }else if(nums[mid] < target){
                start = mid + 1;
            }
            mid = start + (end - start)/2;
        }
        return ans;
    };
